src/validation.cpp

```diff
@@ -3215,26 +3222,31 @@ bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<cons
         if (ret) {
 
             //Do 51% transmitance check, if it triggers, add a badpoint and ban the peer
-            if(std::share_ptr<CBlockIndex> theBlock = GetPreviousBlock(pblock, 5))
-            if(pblock.nTime - theBlock.nTime < (60*10)) {
-                defenseDelayActive = true;
-                time(&defenseStartTime);
-                CNode* pfrom = state.pfrom;
-                //If the block being accepted isn't local
-                if (pfrom && pfrom->addr.ToString().find("local") == std::string::npos && pfrom->addr.ToString().find("127.0.0.") == std::string::npos) {
-                    //We blacklist this block
-                    Checkpoints::AddBadPoint(chainActive.Height(), hash);
-
-                    //Schedule checkpoint 12 blocks from now
-                    checkpointBlockNum = chainActive.Height() + 12;
-
-                    //If so then we ban them
-                    state.DoS(100, false, REJECT_INVALID, "bad-time", true, "51% defense triggered");
-
-                    GetMainSignals().BlockChecked(*pblock, state);
-                    return error("%s: AcceptBlock FAILED - 51% defense triggered", __func__);
-                } else {
-                    return error("%s: AcceptBlock FAILED - Attempted to mine block with disallowed timestamp. Timestamp violates defence rules.", __func__);
+            if(chainActive.Height() > chainparams.GetConsensus().octoberFork)
+            {
+                if(CBlockIndex * theBlock = GetPreviousBlock(*pblock, 5))
+                {
+                    if(pblock->nTime - theBlock->nTime < (60*10)) {
+                        defenseDelayActive = true;
+                        time(&defenseStartTime);
+                        //CNode* pfrom = (CNode*)state.pfrom;
```

Comment:
This line was commented out because `state.pfrom` is always null.  `state` is created on the stack in this function.  This is why it is better to have Cnode * pfrom as a parameter of the function.